device/common/wifi/wifi.h

```diff
@@ -0,0 +1,27 @@
+// Copyright Antioch. All rights reserved.
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace hal {
+namespace wifi {
+
+enum wifi_err_t : int16_t {
+  RES_NOOP = -1,
+  RES_OK = 0,
+};
+
+wifi_err_t wifi_hal_init();
```

Comment:
If you are supposed to check return codes for these, consider using `[[ nodiscard ]]`